include/RAJA/util/View.hpp

```diff
@@ -111,6 +111,163 @@ struct View {
   }
 };
 
+
+// select certain indices from a tuple, given a curated index sequence
+// returns linear index of layout(ar...)
+template <typename Lay, typename Tup, camp::idx_t... Idxs>
+RAJA_HOST_DEVICE RAJA_INLINE 
+RAJA::Index_type selecttuple( Lay lyout, Tup&& tup, camp::idx_seq<Idxs...> )
+{ 
+  return lyout(
+                camp::get<Idxs>(std::forward<Tup>(tup))...
+              );
+}
+
+// sequence combiner
+template <typename Seq1, typename Seq2>
+struct cat_seq;
+
+template <camp::idx_t... Idxs1, camp::idx_t... Idxs2>
+struct cat_seq  < camp::idx_seq<Idxs1...>,
+                  camp::idx_seq<Idxs2...>
+                >
+{
+  using type = camp::idx_seq<Idxs1..., Idxs2...>;
+};
+
+template <typename Seq1, typename Seq2>
+using cat_seq_t = typename cat_seq<Seq1, Seq2>::type;
+
+// sequence offsetter
+template <camp::idx_t Offset, typename Seq>
+struct offset_seq;
+
+template <camp::idx_t Offset, camp::idx_t... Idxs>
+struct offset_seq<Offset, camp::idx_seq<Idxs...>>
+{
+  using type = camp::idx_seq<(Idxs+Offset)...>;
+};
+
+template <camp::idx_t Offset, typename Seq>
+using offset_seq_t = typename offset_seq<Offset, Seq>::type;
+
+// remove the Nth index in a parameter pack
+// returns linear index of layout(ar...)
+template <typename Lay, RAJA::Index_type Nth = 0, typename Tup>
+RAJA_HOST_DEVICE RAJA_INLINE auto removenth( Lay lyout, Tup&& tup ) ->
+  decltype( selecttuple<Lay>(
+              lyout,
+              std::forward<Tup>(tup),
+              cat_seq_t<  camp::make_idx_seq_t<Nth>,  // sequence up to Nth
+                          offset_seq_t<
+                            Nth+1,
+                            camp::make_idx_seq_t<camp::tuple_size<Tup>::value - Nth-1>
+                          > // sequence after Nth
+                       >{}
+            )
+          )
+{
+  return selecttuple<Lay>(
+              lyout,
+              std::forward<Tup>(tup),
+              cat_seq_t<  camp::make_idx_seq_t<Nth>,  // sequence up to Nth
+                          offset_seq_t<
+                            Nth+1,
+                            camp::make_idx_seq_t<camp::tuple_size<Tup>::value - Nth-1>
+                          > // sequence after Nth
+                       >{}
+          );
+}
+
+
+
+
+// P2Pidx represents the pointer-to-pointer index for swizzling of indices.
+// Default of 0 means that the p2p index is in the 0th position.
+template <typename ValueType,
+          typename LayoutType,
+          RAJA::Index_type P2Pidx = 0,
+          typename PointerType = ValueType **>
+struct MultiView {
+  using value_type = ValueType;
+  using pointer_type = PointerType;
+  using layout_type = LayoutType;
+  using nc_value_type = typename std::remove_const<value_type>::type;
+  using nc_pointer_type = 
+    typename std::add_pointer<
+      typename std::add_pointer<
+        typename std::remove_const<
+          typename std::remove_pointer<
+            typename std::remove_pointer<pointer_type>::type
```

Comment:
This is correct, but wow is it a lot of nested `typename ... ::type`.  I have a shorter version of this stuff in camp, suppose it should use the standard names, but this really makes me wish we could get on 14 so at least it could be `*_t` instead of all that gak.

Reply:
@trws Where should I look in CAMP for collapsing this?

Reply:
It's in `helpers.hpp` I think, but apparently I need to add the pointer variants.  It would look something like this:

```c++
using camp::type::ptr;
using camp::type::c;
ptr::add<c::rem<ptr::rem<ptr::rem<pointer_type>>>;
```

I can actually add in the standard names, would be a good thing for discoverability and wouldn't take long at all, then at least you could use the _t variants and it would be less extra syntax but still readable.  The original camp version was put together to be dense, I was using it all over the place in the old nested implementation and needed to be able to chain them.

Reply:
Also, you probably want to remove cv right, or even cvref? Should it preserve volatile or reference qualification?

Reply:
Ok, I made a PR in a CAMP fork to add camp::type::ptr here https://github.com/LLNL/camp/pull/42. It's very simple, just the same as camp::type::ref. It works on MultiView and View as well, and I've used camp::type::cv::rem (but not yet committed due to the CAMP change).

I hope I'm understanding reference qualification correctly, but we're requiring a ** to be passed in, so it should just require non-reference qualification. Because we're expecting **, can we get away with not preserving volatility? Otherwise, I'll need to overload the struct with a volatile version.

Reply:
Hm... I think you are, but now that I'm looking more closely at how this is used, I'm not sure what purpose the result is serving.  It looks like this is being used to have a copy constructor that will allow a `MultiView` with a non-const `value_type` to be used to construct a `MultiView` with a `const` `value_type`.  It seems reasonable to allow assignment from either const or volatile qualified versions into non qualified ones, so it might be good to just do `decay` rather than `remove_const` or `cv::rem` if you want to just eat the const and volatile rather than everything.

Reply:
Sorry, nearly forgot about this comment. Yes, View has the ability to copy construct a non-const `value_type` into a View with `const value_type`, so I just tried to preserve that capability. If we use `std::decay` on the pointer_type, that could potentially convert an array into a pointer, which might be ok considering that a `**` is passed into MultiView and a `*` is passed into View. I can certainly use `decay` on the `value_type` though.